### DISCUSSION TITLE
how do we know position of imu plugin in alice_ros_2 example

### DISCUSSION THREAD

**Discussion by rua0ra1** [cyberbotics/wrestling#40] [Q&A] [answered]
Hi,
I am trying to understanding IMU plugin used in alice_ros_2 example using urdf file. Usually, In gazebo, we add this sensor plugins to robot urdf file and we can also define the position of the sensors. But, in that example, only imu plugin is defined in the urdf file. I am wondering how do we know imu position. Or all the sensor position on the robot are fixed, we are just enabling the ros2 to access the sensor data using those plugins. I am bit confused. I really appreciate if you can answer this.

**Comment by Jean-Eudes-le-retour** [comment, marked as answer]
> all the sensor position on the robot are fixed, we are just enabling the ros2 to access the sensor data using those plugins.

This is correct. The sensors positions are defined inside the robot's model. You can find more information on the position of the sensors on these links:
- https://cyberbotics.com/doc/guide/nao
- [Dedicated proto page](https://webots.cloud/run?version=R2023b&url=https%3A%2F%2Fgithub.com%2Fcyberbotics%2Fwebots%2Fblob%2Fdevelop%2Fprojects%2Frobots%2Fsoftbank%2Fnao%2Fprotos%2FNao.proto&type=undefined)

**Reply by rua0ra1** [reply]
Thank you @Jean-Eudes-le-retour

**Reply by rua0ra1** [reply]
I have one more question related to this. Currently, I can publish camera, led and sonar sensor data as ros2 topics. But, I tried to access the touchdown sensor (robot bumper sensor) on the robot by using type name "Bumper" in the plugin. But, I couldn't see any topic related to this. I am wondering whether I can access all the sensor on the robot by using ros2 or I can only access sensors  given in the ros2 reference sensors (https://github.com/cyberbotics/webots_ros2/wiki/References)

**Reply by Jean-Eudes-le-retour** [reply]
We have answered a similar question [here](https://robotics.stackexchange.com/questions/24416/how-to-add-a-new-sensor-to-webots-ros2-plugins). There are several solutions:
- The simplest: you can modify the [nao_controller.py](https://github.com/cyberbotics/wrestling-alice-ros-2/blob/main/controllers/participant/participant/nao_controller.py) to create topics using the ROS Client Library for Python (rclpy) which will retrieve the values of the various sensors needed using the standard Webots Python API via the [webots_node.robot](https://github.com/cyberbotics/wrestling-alice-ros-2/blob/main/controllers/participant/participant/nao_controller.py#L6) object
- Follow the instructions in [this answer](https://robotics.stackexchange.com/a/24425) and improve the official webots_ros2 package to add more sensors to the supported list.
